### ISSUE TITLE
Global ingestion rate limit inappropriately limits as a cluster scales.

### ISSUE THREAD

**Issue by slim-bean** [grafana/loki#5988]
ingestion rate limits are applied at the distributor, traffic is sent in a round robin fashion to the distributors with the assumption that all traffic is effectively evenly spread over all the distributors.

Based on that assumption when the distributor rate limit is set to "global" the tenant limit is taken and divided by the number of distributors in the cluster. This should result in an effective limit over all the distributors equal to the tenants limit.

As an example, if the tenant limit is 2MB/s, and you have 10 distributors, each distributor would allow 200kB/s of throughput to add up to a combined 2MB/s

The potential problem here is what happens if you have 100 distributors, now each distributor will only allow 20kB/s of throughput.

Because a single push request is only handled by one distributor here's where this can go wrong:

The default max batch size in Promtail is 1MB, meaning it's impossible for this tenant to send full batches. Even if they only send 1 batch per second which is actually a throughput of 1MB/s and is technically below their limit.

In fact in the second case any tenant with a 2MB/sec limit could only send 20kB per batch or less per second before getting rate limited, and the bigger the cluster gets the lower this limit becomes.

**Comment by splitice** [comment]
10 distributors is quite a lot.

We process 200 - 300GB/day (with peaks equivilent to 10-20x average loads) with 3 distributors (primarily to enable quick recovery). We only need 2 of these. How much data are you processing that 10 distributors are required?

Have you considered just scaling up your distributor VMs, servers or pods?

**Comment by slim-bean** [comment]
There are a few reasons we run more distributors, mainly larger ingest volume and smaller pods are easier to pack into nodes for better resource utilization.  

We run our distributors with 500 mCPU and 1G RAM making them easy to pack into open space on nodes so we can get better resource utilization out of our kubernetes environments, basically it's easier to play kubernetes tetris with smaller pieces.

As for scale, our larger clusters continue to grow and some are pushing 50TB a day which puts them in the 500MB/s range and about 10 thousand requests per second (lots of tenants). So this has us running 50-70 distributors to keep the p99 latency sub 50ms and keep our CPU usage close to those request numbers.

The other consideration here is the work we've been doing on the SSD mode where the `write` target is 1:1 with a distributor in each node with an ingester. 

TL;DR your mileage may vary, if your environment favors running more vertical scaled distributors you can definitely do so and run fewer and not have to worry about this so much, but in environments where smaller distributor pods are desirable and scale gets bigger this problem becomes much more impactful!

**Comment by JStickler** [comment]
Closing as stale. This issue was opened against a version of Loki that is no longer supported. If this issues still exists in more recent versions of Loki, please leave a comment and we can reopen the issue.
